packages/code-du-travail-frontend/src/outils/common/InfosCCn.js

```diff
@@ -15,30 +16,17 @@ function StepInfoCCn({ form }) {
   const [ccInfo, setCcInfo] = useConventionState({});
   useEffect(() => {
     form.batch(() => {
-      [
-        "contract",
-        "criteria",
-        "missionFormation",
-        "ruptureContratFauteGrave",
-        "propositionCDIFinContrat",
-        "refusSouplesse",
-        "finContratPeriodeDessai",
-        "propositionCDIFindeContrat",
-        "refusCDIFindeContrat",
-        "interruptionFauteGrave",
-        "refusRenouvellementAuto",
-        "typeRemuneration",
-        "salaire",
-        "salaires"
-      ].forEach(key => form.change(key, undefined));
-      form.change("ccn", ccInfo.convention);
+      Object.keys(form.getState().values)
+        .filter(key => key !== CCN)
+        .forEach(key => form.change(key, undefined));
+      form.change(CCN, ccInfo.convention);
     });
   }, [ccInfo, form]);
   return (
     <>
       <Field
         name="ccn"
```

Comment:
if we use a constant for the CCN key name, we should use it here too (and probably in other place)

Reply:
Done